### PR TITLE
Kava app

### PR DESCRIPTION
**Pull request by fedekunze**
- adds additional parsing methods (base denom -> ticker) for the following token denoms:
  - `ukava -> KAVA`
  - `usdx -> USDX`
  - `uatom -> ATOM`
  - `bnb -> BNB`
- Modifies the app name and version to "Kava" and "1.0.0" respectively
- Add custom transaction parsing for all Kava messages

TODO: 

- [x] Incorporate the .png and .gif logos for Ledger Live and Ledger device respectively

### CODE REVIEW

File: app/Makefile [hunk 1]
```diff
@@ -25,12 +25,12 @@ MY_DIR := $(dir $(lastword $(MAKEFILE_LIST)))
...
-APPNAME = "Cosmos"
-APPVERSION_M=2
-APPVERSION_N=16
+APPNAME = "Kava"
+APPVERSION_M=1
+APPVERSION_N=0
 APPVERSION_P=0
 
-APPPATH = "44'/118'"
+APPPATH = "44'/118'"  # TODO: update ?
```
Comment (fedekunze) [inline app/Makefile]: I think there was an issue with this hdpath

File: app/src/coin.h [hunk 2]
```diff
@@ -16,43 +16,58 @@
...
-#define COIN_DEFAULT_DENOM_FACTOR           6
+    // In non-expert mode, the app will convert from denom base to denom repr
+
+#define KAVA_DENOM_BASE "ukava"
+#define KAVA_DENOM_REPR "KAVA"
+#define KAVA_DENOM_FACTOR 6
+
+#define USDX_DENOM_BASE "usdx"
+#define USDX_DENOM_REPR "USDX"
+#define USDX_DENOM_FACTOR 2
```
Comment (fedekunze) [inline app/src/coin.h]: Idk if this is accurate

File: app/src/coin.h [hunk 3]
```diff
@@ -16,43 +16,58 @@
...
+
+#define ATOM_DENOM_BASE "uatom"
+#define ATOM_DENOM_REPR "ATOM"
+#define ATOM_DENOM_FACTOR 6
+
+#define BNB_DENOM_BASE "bnb"
+#define BNB_DENOM_REPR "BNB"
+#define BNB_DENOM_FACTOR 6
```
Comment (fedekunze) [inline app/src/coin.h]: ditto

File: app/src/coin.h [hunk 4]
```diff
@@ -16,43 +16,58 @@
...
+#define CRYPTO_BLOB_SKIP_BYTES 0
+#define COIN_DEFAULT_CHAINID "kava-3"
 
-// In non-expert mode, the app will convert from uatom to ATOM
-#define COIN_DEFAULT_DENOM_BASE             "uatom"
-#define COIN_DEFAULT_DENOM_REPR             "ATOM"
-#define COIN_DEFAULT_DENOM_FACTOR           6
+    // In non-expert mode, the app will convert from denom base to denom repr
+
+#define KAVA_DENOM_BASE "ukava"
```
Comment (fedekunze) [inline app/src/coin.h:56]: should we add BTC too?

File: app/src/tx_parser.c [hunk 5]
```diff
@@ -50,26 +53,42 @@ __Z_INLINE void strcat_chunk_s(char *dst, uint16_t dst_max, const char *src_chun
...
+    {"bep3/MsgRefundAtomicSwap", "Refund Atomic Swap"},
+    {"cdp/MsgCreateCDP", "Create CDP"},
+    {"cdp/MsgDeposit", "Deposit CDP"},
+    {"cdp/MsgWithdraw", "Withdraw CDP"},
+    {"cdp/MsgDrawDebt", "Draw Debt"},
+    {"cdp/MsgRepayDebt", "Repay Debt"},
+    {"kava/MsgSubmitProposal", "Committee Propose"},
+    {"kava/MsgVote", "Committee Vote"},
+    {"incentive/MsgClaimReward", "Claim Reward"},
+    {"pricefeed/MsgPostPrice", "Post Price"}};
```
Comment (fedekunze) [inline app/src/tx_parser.c]: these are the kava msg replacements

File: app/Makefile (hunk 1 above)
Comment (fedekunze) [reply app/Makefile]:
```suggestion
APPPATH = "44'/459'"
```

File: app/src/coin.h (hunk 2 above)
Comment (fedekunze) [reply app/src/coin.h]:
```suggestion
#define USDX_DENOM_FACTOR 6
```

File: app/src/coin.h (hunk 3 above)
Comment (fedekunze) [reply app/src/coin.h]:
```suggestion
#define BNB_DENOM_FACTOR 8
```

File: app/src/tx_parser.c [hunk 6]
```diff
@@ -50,26 +53,42 @@ __Z_INLINE void strcat_chunk_s(char *dst, uint16_t dst_max, const char *src_chun
...
+    {"cosmos-sdk/MsgSubmitProposal", "Propose"},
+    {"cosmos-sdk/MsgDeposit", "Deposit"},
+    {"cosmos-sdk/MsgVote", "Vote"},
+    {"cosmos-sdk/MsgWithdrawDelegationReward", "Withdraw Reward"},
+    {"auction/MsgPlaceBid", "Bid"},
+    {"bep3/MsgCreateAtomicSwap", "Create Atomic Swap"},
+    {"bep3/MsgRefundAtomicSwap", "Refund Atomic Swap"},
+    {"bep3/MsgClaimAtomicSwap", "Claim Atomic Swap"},
+    {"bep3/MsgCreateAtomicSwap", "Create Atomic Swap"},
+    {"bep3/MsgRefundAtomicSwap", "Refund Atomic Swap"},
```
Comment (fedekunze) [inline app/src/tx_parser.c]:
```suggestion
```

File: app/src/coin.h (hunk 4 above)
Comment (denalimarsh) [reply app/src/coin.h:56]: Add BUSD, BTC-B, USDT, USDC

File: app/src/tx_parser.c [hunk 7]
```diff
@@ -50,26 +53,40 @@ __Z_INLINE void strcat_chunk_s(char *dst, uint16_t dst_max, const char *src_chun
...
+    {"cosmos-sdk/MsgDeposit", "Deposit"},
+    {"cosmos-sdk/MsgVote", "Vote"},
+    {"cosmos-sdk/MsgWithdrawDelegationReward", "Withdraw Reward"},
+    {"auction/MsgPlaceBid", "Bid"},
+    {"bep3/MsgCreateAtomicSwap", "Create Atomic Swap"},
+    {"bep3/MsgRefundAtomicSwap", "Refund Atomic Swap"},
+    {"bep3/MsgClaimAtomicSwap", "Claim Atomic Swap"},
+    {"cdp/MsgCreateCDP", "Create CDP"},
+    {"cdp/MsgDeposit", "Deposit CDP"},
+    {"cdp/MsgWithdraw", "Withdraw CDP"},
```
Comment (denalimarsh) [inline app/src/tx_parser.c]:
```suggestion
    {"cdp/MsgDeposit", "Deposit Collateral"},
    {"cdp/MsgWithdraw", "Withdraw Collateral"},
```

File: app/src/tx_parser.c [hunk 8]
```diff
@@ -50,26 +53,40 @@ __Z_INLINE void strcat_chunk_s(char *dst, uint16_t dst_max, const char *src_chun
...
-};
+    {"cosmos-sdk/MsgSend", "Send"},
+    {"cosmos-sdk/MsgDelegate", "Delegate"},
+    {"cosmos-sdk/MsgUndelegate", "Undelegate"},
+    {"cosmos-sdk/MsgBeginRedelegate", "Redelegate"},
+    {"cosmos-sdk/MsgSubmitProposal", "Propose"},
+    {"cosmos-sdk/MsgDeposit", "Deposit"},
+    {"cosmos-sdk/MsgVote", "Vote"},
+    {"cosmos-sdk/MsgWithdrawDelegationReward", "Withdraw Reward"},
+    {"auction/MsgPlaceBid", "Bid"},
```
Comment (denalimarsh) [inline app/src/tx_parser.c]:
```suggestion
    {"auction/MsgPlaceBid", "Place Bid"},
```

**Review by denalimarsh — DISMISSED** [review]
Looks good, have a couple of additions. Make sure to double-check this feature against the spec at https://docs.google.com/document/d/1TRDw6hWl8TjfFvr6owzBGTiWLX4dZrx4ukhjQ1p7tz0/edit.

Will test using the web app once https://github.com/Kava-Labs/cosmos-ledger is upgraded to match this ledger app's config.

**Pull request merged**